modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,56 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
-import eu.timepit.refined.boolean.Not
+import eu.timepit.refined.boolean.Xor
 import eu.timepit.refined.cats._
-import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.Interval.Open
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
-  case class Amount(
-    value: AmountValue,
-    currency: Currency
-  )
-
-  type AmountValueRefined = Long Refined Not[Equal[0L]]
+  // Only allow values in the closed interval 1 to Long.MaxValue and its negation
+  // Since we have to use literal constants here is visual confirmation:
+  //   From Scala Long:       final val MinValue = java.lang.Long.MIN_VALUE
+  //   From java.lang.Long:   @Native public static final long MIN_VALUE = 0x8000000000000000L;
+  type AmountValuePredicate = Xor[Open[0x8000000000000000L, 0L], Positive]
```

Comment:
I think you can use `And` boolean operation on two intervals:
`(min.val; -1] and [1; max.val)`

it would be simpler and won't require that descriptive reasoning how to get that value

Reply:
@marcinwadon Can you be more specific about what you're looking for here? The syntax you provided does not compile -- is there a way I can specify constants like `Long.MinValue` and get this to work?
Or are you suggesting this, which adds another hard-coded constant:
```
type AmountValuePredicate = Or[OpenClosed[0x8000000000000000L, -1L], Closed[1L, 0x7fffffffffffffffL]]
```

Reply:
I think that you can't use values other than literal - like `Long.MinValue` - because they're not a type, literals like `123` can be interpreted by the compiler as a type. I think that `Not[Equal[0]]` could work.

Reply:
@tbekas actually you can.
@sdavidp here is the snippet: https://scastie.scala-lang.org/kKLmDZDETA2hxbT7Bocz8g

```scala
import eu.timepit.refined._
import eu.timepit.refined.api._
import eu.timepit.refined.boolean._
import eu.timepit.refined.generic._
import eu.timepit.refined.auto._

type ValidLong = Long Refined Not[Or[Or[Equal[W.`Long.MinValue`.T], Equal[W.`0`.T]], Equal[W.`Long.MaxValue`.T]]]

val myLong: ValidLong = -1234L // OK
val invalidLong1: ValidLong = 0L // Compile-time error
val invalidLong2: ValidLong = Long.MaxValue // Compile-time error
val invalidLong3: ValidLong = Long.MinValue // Compile-time error
```

![image](https://user-images.githubusercontent.com/4959931/231798812-08a4f91d-da43-4cd0-9ec5-1a35964328da.png)

Reply:
@marcinwadon can be written in more readable way:

```scala
type ValidLong = Long Refined Not[Equal[W.`Long.MinValue`.T] Or Equal[W.`0`.T] Or Equal[W.`Long.MaxValue`.T]]
```

or even

```scala
type ValidLong = Long Refined Not[
  Equal[W.`Long.MinValue`.T]
  Or Equal[W.`0`.T]
  Or Equal[W.`Long.MaxValue`.T]
]
```

Reply:
I'm good with that @sdavidp :)